src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4640,7 +4640,12 @@ public override IOperation VisitAnonymousObjectCreation(IAnonymousObjectCreation
 
         public override IOperation VisitCollectionElementInitializer(ICollectionElementInitializerOperation operation, int? captureIdForResult)
         {
-            return new CollectionElementInitializerExpression(operation.AddMethod, operation.IsDynamic, VisitArray(operation.Arguments), semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
+            return new CollectionElementInitializerExpression(operation.AddMethod, VisitArray(operation.Arguments), semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
+        }
+
+        public override IOperation VisitDynamicCollectionElementInitializer(IDynamicCollectionElementInitializerOperation operation, int? argument)
+        {
+            return new DynamicCollectionElementInitializerOperation(VisitArray(operation.Arguments), ((Operation)operation).SemanticModel, operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
```

Comment:
Why does this get a semantic model bot non-dynamic collection initializer does not?

Reply:
No longer relevant, but was a bug.

---
In reply to: [187511369](https://github.com/dotnet/roslyn/pull/26781#discussion_r187511369) [](ancestors = 187511369)